grammar.js

```diff
@@ -308,10 +308,16 @@ module.exports = grammar({
             seq('@load', $.file),
             seq('@load-plugin', $.id),
             seq('@load-sigs', $.file),
+            seq('@pragma', choice(seq('push', $.pragma_stack_choices),
+                                  seq('pop', $.pragma_stack_choices))),
```

Comment:
This repeats the pattern for attributes where all currently supported choices get hardcoded in the grammar. Like for attributes, this shouldn't be needed to parse source files and will just introduce maintenance overhead. I'd suggest to add handling for `@pragma`, but allow it to be followed by a whitespace-separated list of values. With that downstream consumers can inspect the pragma payload and e.g., in the case of `zeek-format` fix whitespace among `@pragma`, `push`/`pop` and `ignore-deprecations`.

Given the following input:

```zeek
@pragma push ignore-deprecations
```

with your grammar we get the following CST

```
(source_file
  (decl
    (preproc_directive
      (pragma_stack_choices))))
)
```

We get a node for `pragma_stack_choices`, but nothing for the `@pragma` itself, so consumers would need to manually inspect `preproc_directive` to see it. I'd suggest to expose `@pragma` as its own node (useful information for consumers), but not model `pragma_stack_choices` at all via above argument for maintainability.

I'd suggest

```suggestion
            $.pragma,
```

and then below (not binding `$` since it is unused in the rule, allowing arbitrary value payload)

```js
        pragma: _ => seq('@pragma', repeat1(/\S+/)),
```

With that we would get the following CST for above example:

```
(source_file
  (decl
    (preproc_directive
      (pragma)))
)
```

Reply:
Yeah that's fine with me, will do.